hilti/runtime/include/type-info.h

```diff
@@ -180,7 +180,7 @@ class DereferencableType {
 
 private:
     const TypeInfo* _vtype;
-    const Accessor _accessor;
+    Accessor* _accessor;
```

Comment:
Could this remains `const`?

Also, how about turning the `Accessor` typedef into a `pointer to <function>` (i.e., `using Accessor = const void* (*) (const Value& v);`, then we can pass it around as just `Accessor` just like we do further below for `Accessor` types that are tuples.